lib/rules/no-unused-vars.js

```diff
@@ -466,7 +477,8 @@ module.exports = {
                     (
                         parent.type === "AssignmentExpression" &&
                         parent.left === id &&
-                        isUnusedExpression(parent)
+                        isUnusedExpression(parent) &&
+                        !isLogicalAssignmentOperator(parent.operator)
```

Comment:
```suggestion
                        !astUtils.isLogicalAssignmentOperator(parent.operator)
```

We can use [this helper function](https://github.com/eslint/eslint/blob/c8b1f4d61a256727755d561bf53f889b6cd712e0/lib/rules/utils/ast-utils.js#L757-L759).